component/common/loki/client/batch.go

```diff
@@ -201,3 +201,11 @@ func (b *batch) reportAsSentData(h SentDataMarkerHandler) {
 		h.UpdateSentData(seg, data)
 	}
 }
+
+func entrySize(entry logproto.Entry) int {
+	structuredMetadataSize := 0
+	for _, label := range entry.StructuredMetadata {
+		structuredMetadataSize += label.Size()
+	}
+	return len(entry.Line) + structuredMetadataSize
+}
```

Comment:
nit: do we know what `entry.Parsed` is used for? Should it also be included here?

Reply:
I would like to merge it as-is, because my goal here is to just match the upstream: https://github.com/grafana/loki/blob/main/clients/pkg/promtail/client/batch.go#L150-L156 

But you have a valid question. I wonder why `Parsed` is not included in size or why the existing Size method is not used: https://github.com/grafana/loki/blob/main/pkg/push/types.go#L720 

I think @vlad-diachenko will be able to answer the above.

Reply:
I see. Yeah, I think we should move forward with this